crates/bevy_core_pipeline/src/bloom/bloom.wgsl

```diff
@@ -51,19 +52,33 @@ fn karis_average(color: vec3<f32>) -> f32 {
 
 // [COD] slide 153
 fn sample_input_13_tap(uv: vec2<f32>) -> vec3<f32> {
-    let a = textureSample(input_texture, s, uv, vec2<i32>(-2, 2)).rgb;
-    let b = textureSample(input_texture, s, uv, vec2<i32>(0, 2)).rgb;
-    let c = textureSample(input_texture, s, uv, vec2<i32>(2, 2)).rgb;
-    let d = textureSample(input_texture, s, uv, vec2<i32>(-2, 0)).rgb;
+#ifdef FIRST_DOWNSAMPLE
+    // Prevents noticeable sampling/ghosting artifacts when using large scales.
+    let scale = vec2<f32>(1.0, 1.0);
+#else
+    let scale = uniforms.scale;
+#endif
+    let x_s = scale.x / f32(textureDimensions(input_texture).x);
+    let y_s = scale.y / f32(textureDimensions(input_texture).y);
+    let x_l = 2.0 * x_s;
+    let y_l = 2.0 * y_s;
+    let x_ns = -1.0 * x_s;
+    let y_ns = -1.0 * y_s;
+    let x_nl = -2.0 * x_s;
+    let y_nl = -2.0 * y_s;
+    let a = textureSample(input_texture, s, uv + vec2<f32>(x_nl, y_l)).rgb;
```

Comment:
Note for reviewers - I was not able to use the offset syntax used before, because I was hitting shader compiler errors when trying to pass computed offsets into what it said was a const context.

Reply:
I'm not sure. I'd be curious to learn from others whether there are dedicated hardware paths for offsets when texture fetching on modern gpus. I wonder if it could also affect mips.

The other way we could handle this is by implementing support for [`@override`] (https://www.w3.org/TR/WGSL/#override-decls) although it's worth noting that this would create a new pipeline for each override so wouldn't be suitable for scenarios where you want to animate the bloom scale effect (and would be limiting to the +/-8 of that offset param). 

Assuming there aren't major performance concerns that others are familiar with this makes sense to me.

Reply:
Another option would be to branch on the special case when the offsets == 1, in which case we can take the happy path and use the const offset syntax for sampling. From my understanding, this branching should be cheap, because it is branching on a uniform, meaning all groups will take the same branch.

Reply:
The spec doesn't really mention anything about perf. I just says the offset needs to be a const expression.
https://www.w3.org/TR/WGSL/#texturesample

Reply:
Using a uniform branch is a good idea too, but I'd say let's keep it simple unless there's evidence of performance issue. The main question is always mobile. I'll bench mark on desktop this weekend if someone more knowledgeable doesn't get back with a more definitive answer about hardware.

Reply:
Oops, well, I added the uniform branch. If anything, it will make it easier for you to test if there is any perf impact of the slow path. As soon as the scale is non-uniform, it takes the slow path automatically.

Reply:
Using the shader def is even better. I'd still like to benchmark this (just because I'm curious), but this certainly eliminates any concerns.

Reply:
I am not englishing very well today. Yeah, I added a shader def, figured that would be pretty bulletproof, though it comes at the cost of some extra boilerplate on the setup side.